diff --git a/libs/LuaJitLib/CMakeLists.txt b/libs/LuaJitLib/CMakeLists.txt
index 54182fcd..cfb8105f 100644
--- a/libs/LuaJitLib/CMakeLists.txt
+++ b/libs/LuaJitLib/CMakeLists.txt
@@ -310,8 +310,8 @@ else()
   IF(MINILUA_USE_LIBM)
     set(MINILUA_LIBS m)
   ENDIF()
-  if(LINUX_ON_ARM)
-    message(STATUS "Forcing minilua libm on LINUX_ON_ARM")
+  if(LINUX_ON_ARM OR NOT ("${ANDROID_ABI}" STREQUAL ""))
+    message(STATUS "Forcing minilua libm on LINUX_ON_ARM / ANDROID")
     list(APPEND MINILUA_LIBS m)
   endif()
 
Submodule libs/sst/sst-plugininfra contains modified content
diff --git a/libs/sst/sst-plugininfra/src/paths_linux.cpp b/libs/sst/sst-plugininfra/src/paths_linux.cpp
index 4a1d94e..a6e0b8b 100644
--- a/libs/sst/sst-plugininfra/src/paths_linux.cpp
+++ b/libs/sst/sst-plugininfra/src/paths_linux.cpp
@@ -28,10 +28,14 @@ namespace paths
 {
 fs::path homePath()
 {
+#if ANDROID
+    return fs::path{"/data/data/org.androidaudioplugin.ports.juce.surge_xt"};
+#else
     const char *const path = std::getenv("HOME");
     if (!path || !path[0])
         throw std::runtime_error{"The environment variable HOME is unset or empty"};
     return fs::path{path};
+#endif
 }
 
 fs::path sharedLibraryBinaryPath()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 31bee8d5..6b2aed43 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -164,3 +164,10 @@ endif()
 if(NOT CMAKE_CROSSCOMPILING AND NOT SURGE_SKIP_JUCE_FOR_RACK)
   include(cmake/pluginval.cmake)
 endif()
+
+# begin AAP specifics -->
+
+message("AAP_DIR: ${AAP_DIR}")
+message("AAP_JUCE_DIR: ${AAP_JUCE_DIR}")
+juce_add_modules(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+# <-- end AAP specifics
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 2d5ba615..e6f9eab0 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -306,6 +306,9 @@ if(APPLE)
     "-framework CoreFoundation"
     "-framework Foundation"
   )
+elseif(ANDROID)
+  target_compile_definitions(${PROJECT_NAME} PUBLIC LINUX=1)
+  target_link_libraries(${PROJECT_NAME} PUBLIC surge::simde)
 elseif(UNIX)
   target_compile_definitions(${PROJECT_NAME} PUBLIC LINUX=1)
 
diff --git a/src/surge-fx/CMakeLists.txt b/src/surge-fx/CMakeLists.txt
index ff9c96ae..c94661ce 100644
--- a/src/surge-fx/CMakeLists.txt
+++ b/src/surge-fx/CMakeLists.txt
@@ -95,3 +95,42 @@ if(SURGE_BUILD_CLAP)
 endif()
 
 surge_juce_package(${PROJECT_NAME} "Surge XT Effects")
+
+# begin AAP specifics -->
+
+# They are needed in desktop too, for generate-aap-metadata.
+include_directories(
+        "${AAP_DIR}/include"
+        "${AAP_JUCE_DIR}/modules"
+)
+
+add_definitions([[-DJUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]])
+
+target_compile_definitions(${PROJECT_NAME} PUBLIC
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1
+)
+
+if (ANDROID)
+
+    # dependencies
+    find_library(log "log")
+    find_library(android "android")
+    find_library(glesv2 "GLESv2")
+    find_library(egl "EGL")
+    set(cpufeatures_lib "cpufeatures")
+    set(oboe_lib "oboe")
+
+    target_include_directories(${PROJECT_NAME} PRIVATE
+            "${ANDROID_NDK}/sources/android/cpufeatures"
+            "${OBOE_DIR}/include"
+            )
+
+endif (ANDROID)
+
+target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+        aap_audio_processors
+        )
+# <-- end Android specifics
+# <-- end AAP specifics
diff --git a/src/surge-xt/CMakeLists.txt b/src/surge-xt/CMakeLists.txt
index 25897a35..1bce85de 100644
--- a/src/surge-xt/CMakeLists.txt
+++ b/src/surge-xt/CMakeLists.txt
@@ -257,3 +257,42 @@ surge_juce_package(${PROJECT_NAME} "Surge XT")
 if (${SURGE_BUILD_TESTRUNNER})
   add_subdirectory(xt-tests)
 endif()
+
+# begin AAP specifics -->
+
+# They are needed in desktop too, for generate-aap-metadata.
+include_directories(
+        "${AAP_DIR}/include"
+        "${AAP_JUCE_DIR}/modules"
+)
+
+add_definitions([[-DJUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]])
+
+target_compile_definitions(${PROJECT_NAME} PUBLIC
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1
+)
+
+if (ANDROID)
+
+  # dependencies
+  find_library(log "log")
+  find_library(android "android")
+  find_library(glesv2 "GLESv2")
+  find_library(egl "EGL")
+  set(cpufeatures_lib "cpufeatures")
+  set(oboe_lib "oboe")
+
+  target_include_directories(${PROJECT_NAME} PRIVATE
+          "${ANDROID_NDK}/sources/android/cpufeatures"
+          "${OBOE_DIR}/include"
+  )
+
+endif (ANDROID)
+
+target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+        aap_audio_processors
+)
+# <-- end Android specifics
+# <-- end AAP specifics
