diff --git a/libs/luajitlib/CMakeLists.txt b/libs/luajitlib/CMakeLists.txt
index 83567e18..6970b1d4 100644
--- a/libs/luajitlib/CMakeLists.txt
+++ b/libs/luajitlib/CMakeLists.txt
@@ -58,6 +58,13 @@ if(NOT EXISTS "${LUAJIT_LIBRARY}")
       RESULT_VARIABLE LUAJIT_RESULT
       OUTPUT_VARIABLE LUAJIT_OUTPUT
       ERROR_VARIABLE LUAJIT_ERROR)
+  elseif(ANDROID)
+    message(STATUS "Running ./build-linux-luajit.sh \"${LUAJIT_BUILD_DIR}\" in ${CMAKE_CURRENT_LIST_DIR}")
+    execute_process(WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
+      COMMAND ./build-linux-luajit.sh "${LUAJIT_BUILD_DIR}"
+      RESULT_VARIABLE LUAJIT_RESULT
+      OUTPUT_VARIABLE LUAJIT_OUTPUT
+      ERROR_VARIABLE LUAJIT_ERROR)
   elseif(UNIX)
     message(STATUS "Running ./build-linux-luajit.sh \"${LUAJIT_BUILD_DIR}\" in ${CMAKE_CURRENT_LIST_DIR}")
     execute_process(WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
Submodule libs/sst/sst-plugininfra contains modified content
diff --git a/libs/sst/sst-plugininfra/src/paths_linux.cpp b/libs/sst/sst-plugininfra/src/paths_linux.cpp
index 4a1d94e..a6e0b8b 100644
--- a/libs/sst/sst-plugininfra/src/paths_linux.cpp
+++ b/libs/sst/sst-plugininfra/src/paths_linux.cpp
@@ -28,10 +28,14 @@ namespace paths
 {
 fs::path homePath()
 {
+#if ANDROID
+    return fs::path{"/data/data/org.androidaudioplugin.ports.juce.surge_xt"};
+#else
     const char *const path = std::getenv("HOME");
     if (!path || !path[0])
         throw std::runtime_error{"The environment variable HOME is unset or empty"};
     return fs::path{path};
+#endif
 }
 
 fs::path sharedLibraryBinaryPath()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 840789f5..1af251c4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -171,3 +171,10 @@ endif()
 if(NOT CMAKE_CROSSCOMPILING AND NOT SURGE_SKIP_JUCE_FOR_RACK)
   include(cmake/pluginval.cmake)
 endif()
+
+# begin AAP specifics -->
+
+message("AAP_DIR: ${AAP_DIR}")
+message("AAP_JUCE_DIR: ${AAP_JUCE_DIR}")
+juce_add_modules(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+# <-- end AAP specifics
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index c63a929f..aa71860d 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -326,10 +326,12 @@ elseif(UNIX)
       pthread
     )
   else()
+   if(NOT ANDROID)
     target_link_libraries(${PROJECT_NAME} PRIVATE
       pthread
       -Wl,--no-undefined
     )
+   endif()
   endif()
 
   if(CMAKE_SYSTEM_NAME MATCHES "BSD")
diff --git a/src/surge-fx/CMakeLists.txt b/src/surge-fx/CMakeLists.txt
index bbf675de..deb7a66e 100644
--- a/src/surge-fx/CMakeLists.txt
+++ b/src/surge-fx/CMakeLists.txt
@@ -94,3 +94,42 @@ if(SURGE_BUILD_CLAP)
 endif()
 
 surge_juce_package(${PROJECT_NAME} "Surge XT Effects")
+
+# begin AAP specifics -->
+
+# They are needed in desktop too, for generate-aap-metadata.
+include_directories(
+        "${AAP_DIR}/include"
+        "${AAP_JUCE_DIR}/modules"
+)
+
+add_definitions([[-DJUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]])
+
+target_compile_definitions(${PROJECT_NAME} PUBLIC
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1
+        )
+
+if (ANDROID)
+
+    # dependencies
+    find_library(log "log")
+    find_library(android "android")
+    find_library(glesv2 "GLESv2")
+    find_library(egl "EGL")
+    set(cpufeatures_lib "cpufeatures")
+    set(oboe_lib "oboe")
+
+    target_include_directories(${PROJECT_NAME} PRIVATE
+            "${ANDROID_NDK}/sources/android/cpufeatures"
+            "${OBOE_DIR}/include"
+            )
+
+endif (ANDROID)
+
+target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+        aap_audio_processors
+        )
+# <-- end Android specifics
+# <-- end AAP specifics
diff --git a/src/surge-xt/CMakeLists.txt b/src/surge-xt/CMakeLists.txt
index 4bcd3baa..d409f488 100644
--- a/src/surge-xt/CMakeLists.txt
+++ b/src/surge-xt/CMakeLists.txt
@@ -263,3 +263,43 @@ surge_juce_package(${PROJECT_NAME} "Surge XT")
 if (${SURGE_BUILD_TESTRUNNER})
   add_subdirectory(xt-tests)
 endif()
+
+
+# begin AAP specifics -->
+
+# They are needed in desktop too, for generate-aap-metadata.
+include_directories(
+        "${AAP_DIR}/include"
+        "${AAP_JUCE_DIR}/modules"
+)
+
+add_definitions([[-DJUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]])
+
+target_compile_definitions(${PROJECT_NAME} PUBLIC
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1
+        )
+
+if (ANDROID)
+
+    # dependencies
+    find_library(log "log")
+    find_library(android "android")
+    find_library(glesv2 "GLESv2")
+    find_library(egl "EGL")
+    set(cpufeatures_lib "cpufeatures")
+    set(oboe_lib "oboe")
+
+    target_include_directories(${PROJECT_NAME} PRIVATE
+            "${ANDROID_NDK}/sources/android/cpufeatures"
+            "${OBOE_DIR}/include"
+            )
+
+endif (ANDROID)
+
+target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+        aap_audio_processors
+        )
+# <-- end Android specifics
+# <-- end AAP specifics
